### PR TITLE
Localize permission description

### LINKED ISSUE #10551: Ability to localize permissions
### Is your feature request related to a problem? Please describe.
Currently, the `Name` and `Description` of `Permission`s are not really suited for localization. You can I suppose return `Permission` objects with `Description` strings freshly run through the localizer in your `IPermissionProvider.GetPermissionsAsync()` implementation but that would be awkward to do every time. And for `Name`, you can't do localization because it's a technical identifier.

### Describe the solution you'd like
Ideally, `Permission` would have `LocalizedString` `DisplayName` and `Description` fields that would work together with the rest of the permission ecosystem.

### Describe alternatives you've considered
You could also run the strings through a localizer when displaying them (currently probably only in Edit.cshtml of `OrchardCore.Roles`). Thus, you could hand-build the necessary PO file.

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #10551

### CODE REVIEW

**Comment by MikeAlhayek** [comment]
I don't know why we need to localize the description each time we initialize the class. We really only need to localize the description, only when description will be rendered on the UI. When calling the `GetPermissionsAsync()`, we don't always display the description. In many places we call the `GetPermissionsAsync()` method just to get permission names. I think the best place to call the localizer is the Roles UI. Editing a role is the only place we show the description property. It should be a good place to localize the description before we show them there. I understand that description is a variable and may not be a good idea to translate the variable, but in this case I think it should be fine. Otherwise we are translating description all the time even if we are not planing to use the translation.

**Comment by jtkech** [comment]
I agree with both ;)

For the po extractor tool should appear explicitly as `S["some string"]` without using a variable.

But yes better to be lazily executed as needed.

Maybe a new overridable `LocalizedDescription` prop that would be equal to `Description` by default.

**Comment by MikeAlhayek** [comment]
If we want a lazy localizer, we could add a method to Permission class `GetLocalizedDecription(IStringLocalizer s)` and that would return `s[Description]`.  But at the same time, I am not sure if the added complicity is needed since this is something we only use in one place (Roles.Admin.Edit).

In addition to calling `S[permission.Description]` in the during editing a role, we could also remove the public setter from description property, just to ensure that description is read-only.

**Comment by hishamco** [comment]
> For the po extractor tool should appear explicitly as S["some string"] without using a variable.

That's why I made it simple, so no need to change anything in PO Extractor

> Maybe a new overridable LocalizedDescription prop that would be equal to Description by default.

We could, but I'm not sure why we need to have both localized and non localized

**Comment by hishamco** [comment]
What if we use the lazy loading like this

```csharp
    private static Lazy<Permission> _manageCultures;

    private readonly IStringLocalizer S;

    public Permissions(IStringLocalizer<Permissions> localizer)
    {
        S = localizer;
        _manageCultures = new Lazy<Permission>(() => new("ManageCultures", S["Manage supported culture"]));
    }

    public static Permission ManageCultures => _manageCultures.Value;
```

File: src/OrchardCore.Modules/OrchardCore.Localization/PermissionProvider.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+namespace OrchardCore.Localization;
+
+/// <summary>
+/// Represents the localization module permissions.
+/// </summary>
+public class PermissionProvider : IPermissionProvider
+{
+    public PermissionProvider(IStringLocalizer<PermissionProvider> S)
+    {
+        Permissions.ManageCultures = new Permission(nameof(Permissions.ManageCultures), S["View audit trail"]);
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Localization/PermissionProvider.cs:15]: @sebastienros If this style is acceptable, I will loop through every permissions in the code-bae

**Pull request closed without merging**